solv/tdnfquery.c

```diff
@@ -505,10 +505,7 @@ SolvGenerateCommonJob(
     {
         queue_empty(&queueJob);
         queue_push2(&queueJob, SOLVER_SOLVABLE_ALL, 0);
-        if (pQuery->queueRepoFilter.count)
-        {
-            selection_filter(pPool, &queueJob, &pQuery->queueRepoFilter);
-        }
+        selection_filter(pPool, &queueJob, &pQuery->queueRepoFilter);
```

Comment:
Why are you removing the `if (...` here?

Reply:
We enter inside this block based on same condition, so it's redundant.